Initialtask/tests/Entity/ProductTest.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace App\Entity;
+
+class ProductTest extends \PHPUnit\Framework\TestCase
+{
+    private function createTestProduct()
+    {
+        $productData = [
+            'Product Name' => 'tName',
+            'Product Code' => 'tCode',
+            'Product Description' => 'tDesc',
+            'Cost in GBP' => 25.3,
+            'Stock' => 8,
+            'Discontinued' => 'yes',
+        ];
+
+        $product = new Product($productData);
+        $product->setAdded();
+
+        return $product;
+    }
+
+    private function callMethod($object, string $method, array $parameters = [])
+    {
+        try {
+            $className = get_class($object);
+            $reflection = new \ReflectionClass($className);
+        } catch (\ReflectionException $e) {
+            throw new \Exception($e->getMessage());
+        }
+
+        $method = $reflection->getMethod($method);
+        $method->setAccessible(true);
+
+        return $method->invokeArgs($object, $parameters);
+    }
+
+    public function testGetCost()
+    {
+        $this->assertIsFloat($this->createTestProduct()->getCost());
+    }
+
+    public function testGetName()
+    {
+        $this->assertIsString($this->createTestProduct()->getName());
+    }
+
+    public function testGetCode()
+    {
+        $this->assertIsString($this->createTestProduct()->getCode());
+    }
+
+    public function testGetStock()
+    {
+        $this->assertIsInt($this->createTestProduct()->getStock());
+    }
+
+    public function testGetDescription()
+    {
+        $this->assertIsString($this->createTestProduct()->getDescription());
+    }
+
+    public function testSetDescription()
+    {
+        $newDesc = 'newDescription';
```

Comment:
Why don't you set properties during object initialization, you won't need reflection in this case

Reply:
Casue I checking exactly set/get methods, not a __constructor